src/file.js

```diff
@@ -745,26 +776,42 @@ File.prototype.createReadStream = function(options) {
  * });
  */
 File.prototype.createResumableUpload = function(options, callback) {
+  var self = this;
+
   if (is.fn(options)) {
     callback = options;
     options = {};
   }
 
-  resumableUpload.createURI(
-    {
-      authClient: this.bucket.storage.authClient,
-      bucket: this.bucket.name,
-      file: this.name,
-      generation: this.generation,
-      metadata: options.metadata,
-      origin: options.origin,
-      predefinedAcl: options.predefinedAcl,
-      private: options.private,
-      public: options.public,
-      userProject: options.userProject,
-    },
-    callback
-  );
+  this.storage.authClient.getProjectId(function(err, projectId) {
+    // Ignore error.
+
+    if (self.kmsKeyName) {
+      self.kmsKeyName = common.util.replaceProjectIdToken(
+        self.kmsKeyName,
+        projectId
+      );
+    }
```

Comment:
Why is this kmsKeyName doing a replacement for projectId?

Reply:
The format of KMS key name that we're anticipating is:

```
`projects/my-project/locations/location/keyRings/my-kr/cryptoKeys/my-key`
```

Our library doesn't typically require the user to specify their project ID, which is why I was doing the replacement. However, if we're always requiring this full resource path, then it would be highly unlikely that the user wouldn't be providing their own project ID. That was an oversight on my part -- removed :)